### PR TITLE
Rainbow tunic

### PR DESCRIPTION
**Pull request by rrealmuto**
Because people asked me to PR this. Does what the title suggests.

Video showing the rainbow effect
https://user-images.githubusercontent.com/5086838/235819047-46b4eb98-9afe-4ae4-96a1-41591518911c.mp4

Screenshot of the equipment screen icons when rainbow is selected (it's selected for Kokiri and Goron tunic in this seed).
rainbow tunic equipment screenshot

### CODE REVIEW

**Comment by rrealmuto** [comment]
Things I'm sure people will complain about:

- [ ] No way to change the cycling speed
- [x] The icons in the equipment menu are the default colors if Rainbow is selected for a particular tunic

File: Colors.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def get_tunic_colors():
 
 
 def get_tunic_color_options():
-    return meta_color_choices + get_tunic_colors()
+    return meta_color_choices + ["Rainbow"] + get_tunic_colors()
```
Comment (rrealmuto) [inline Colors.py:256]: This will probably break if the "Random Choice" option selects Rainbow

File: Cosmetics.py [hunk 2]
```diff
@@ -91,20 +91,34 @@ def patch_tunic_icon(rom, tunic, color):
 
 def patch_tunic_colors(rom, settings, log, symbols):
     # patch tunic colors
+    
+    tunic_address = rom.sym("CFG_TUNIC_COLORS")
```
Comment (Cuphat) [inline Cosmetics.py]: `rom.sym` in cosmetics code is unsafe as patching a patch file from a different version would write to the wrong location.

Comment (rrealmuto) [reply Cosmetics.py]: Should be fixed now.

Comment (rrealmuto) [reply Cosmetics.py]: Though it probably will break on older patch versions.

Comment (rrealmuto) [reply Cosmetics.py]: Ok now I think it won't break.

**Comment by Cuphat** [comment]
These are my proposed changes for version compatibility. You pushed yours as I was working on it. It's a softer fallthrough for rainbow tunics (similar to how rainbow navi behaves). https://github.com/Cuphat/OoT-Randomizer/commit/1903c0d936e80b63d4f96f6ee25694960d7e3704

**Review by cjohnson57 — APPROVED** [review]

**Comment by ETR-BTF** [comment]
Apparently fixes #751, #1698: https://discord.com/channels/274180765816848384/438697911753113600/1100620422153777152

**Comment by fenhl** [comment]
Does that mean we can get rid of the color restriction for glitched/no logic?

**Comment by rrealmuto** [comment]
I thought I already did. But yeah it seems to fix the bug. I can't weirdshot myself but I had someone who could make it consistently crash with the wrong color, and after the patch it wouldn't crash

**Comment by fenhl** [comment]
Oh, so you did.

**Pull request merged**